sql/testdata/datetime

```diff
@@ -72,3 +82,175 @@ query B
 SELECT now() + '1m'::interval > now(), now() + '1m'::interval >= now()
 ----
 true true
+
+# TIMESTAMP/DATE builtins.
+
+query T
+SELECT age(timestamp '2001-04-10 22:06:45', timestamp '1957-06-13')
+----
+384190h6m45s
+
+query T
+SELECT age(timestamp '1957-06-13') - age(now(), timestamp '1957-06-13') < interval '10s'
+----
+true
+
+query T
+SELECT now() - timestamp '2015-06-13' > interval '100h'
+----
+true
+
+query T
+SELECT now() - current_date()::timestamp < interval '24h10s'
+----
+true
+
+query T
+SELECT current_date() - current_date() = interval '0s'
+----
+true
+
+query error .*current_date.*
+SELECT current_date
+
+query error .*current_timestamp.*
+SELECT current_timestamp
+
+query T
+SELECT now() - current_timestamp() = interval '0s'
+----
+true
+
+query T
+SELECT now() - statement_timestamp() = interval '0s'
+----
+true
+
+query T
+SELECT clock_timestamp() - statement_timestamp() < interval '10s'
+----
+true
+
+query T
+SELECT now() - transaction_timestamp() = interval '0s'
+----
+true
+
+statement ok
+BEGIN TRANSACTION
+
+statement ok
+CREATE TABLE kv (
+  k CHAR PRIMARY KEY,
+  v TIMESTAMP
+)
+
+statement ok
+INSERT INTO kv (k,v) VALUES ('a', transaction_timestamp())
+
+query I
+SELECT k FROM kv
+----
+a
+
+query I
+SELECT k FROM kv where v = transaction_timestamp()
+----
+a
+
+statement ok
+COMMIT TRANSACTION
+
+# Test that the transaction_timestamp can differ from the current_timestamp.
+# Check that the transaction_timestamp changes with each transaction.
+# We use, SELECT * FROM kv, to insert delays.
+statement ok
+BEGIN; INSERT INTO kv (k,v) VALUES ('b', transaction_timestamp()); SELECT * FROM kv; INSERT INTO kv (k,v) VALUES ('c', transaction_timestamp()); SELECT * FROM kv; INSERT INTO kv (k,v) VALUES ('d', current_timestamp()); SELECT * FROM KV; COMMIT; BEGIN; SELECT * FROM KV; INSERT INTO kv (k,v) VALUES ('e', transaction_timestamp()); COMMIT; 
```

Comment:
FYI, this does not need to be one line. The logic-test code reads statements to execute up until the next blank line. That's how the `CREATE TABLE` at line 143 works.